### DISCUSSION TITLE
How to use For each with createResource

### DISCUSSION THREAD

**Discussion by ezcamlive** [solidjs/solid#737] [Q&A] [answered]
const [data, { refetch }] = createResource<Products>(fetchProducts)
<For each={data} fallback={<div>Loading...</div>}>
                {(item) => <div>{item()}</div>}
 </For>

For each shows this error
Type 'Resource<Products>' is not assignable to type 'false | readonly unknown[]'.
  Type 'Resource<Products>' is missing the following properties from type 'readonly unknown[]': concat, join, slice, indexOf, and 18 more.

**Comment by lxsmnsyc** [comment, marked as answer]
You should call `data()`

```js
<For each={data()}>
  {...}
</For>
```

**Comment by ezcamlive** [comment]
Thanks. It works. I got help on Discord. https://playground.solidjs.com/?hash=-1089171677&version=1.2.3
